core.go

```diff
@@ -89,19 +129,20 @@ func (c *core) Write(ent zapcore.Entry, fs []zapcore.Field) error {
 		event.Message = ent.Message
 		event.Timestamp = ent.Time
 		event.Level = sentrySeverity(ent.Level)
-		event.Extra = clone.fields
 		event.Tags = c.cfg.Tags
 		event.Exception = clone.createExceptions()
 
 		if event.Exception == nil && !c.cfg.DisableStacktrace && c.client.Options().AttachStacktrace {
 			stacktrace := sentry.NewStacktrace()
 			if stacktrace != nil {
-				stacktrace.Frames = filterFrames(stacktrace.Frames)
+				stacktrace.Frames = filterFrames(stacktrace.Frames, c.cfg.FrameFilters)
 				event.Threads = []sentry.Thread{{Stacktrace: stacktrace, Current: true}}
 			}
 		}
 
-		_ = c.client.CaptureEvent(event, nil, c.scope())
+		s := c.scope()
+		s.SetContext("fields", clone.fields)
```

Comment:
Yeah, I wouldn't merge this within "Add frame filtering functionality" (and other changes related to the Extra/Context). I think you should split this into two PRs.